rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1119,7 +1131,15 @@ void main_window::CreateConnects()
 		connect(&dlg, &settings_dialog::GuiSettingsSyncRequest, this, &main_window::ConfigureGuiFromSettings);
 		connect(&dlg, &settings_dialog::GuiStylesheetRequest, this, &main_window::RequestGlobalStylesheetChange);
 		connect(&dlg, &settings_dialog::GuiRepaintRequest, this, &main_window::RepaintGui);
-		dlg.exec();
+		if (dlg.exec() == QDialog::Accepted && !guiSettings->GetValue(gui::m_richPresence).toBool() && m_richPresence)
+		{
+			// Discord Rich Presence Integration
+#ifdef WITH_DISCORD_RPC
+			DiscordRichPresence discordPresence = {};
+			Discord_UpdatePresence(&discordPresence);
```

Comment:
Why are you not setting
`discordPresence.largeImageKey` and `discordPresence.largeImageText` here?

Reply:
Also i think using a connect for accepted might be a bit more fitting here.

Reply:
The idea is to clear as much rich presence as possible, that's why im sending without `largeImageKey` and `largeImageText`. Also I tried using accepted connect, but it didn't work, I even asked sloth about it. He came up with the current solution. Anyway this code will probably be removed as I think the best course of action is to require a restart after changing the option.